pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -67,19 +67,34 @@ func init() {
 var (
 	gracefulDeletionVersion = utilversion.MustParseSemantic("v1.1.0")
 
-	// UnreachableTaintTemplate is the taint for when a node becomes unreachable.
-	UnreachableTaintTemplate = &v1.Taint{
+	// UnreachableNoExecuteTaintTemplate is the taint with effect NoExecute for
+	// when a node becomes unreachable.
+	UnreachableNoExecuteTaintTemplate = &v1.Taint{
 		Key:    algorithm.TaintNodeUnreachable,
 		Effect: v1.TaintEffectNoExecute,
 	}
 
-	// NotReadyTaintTemplate is the taint for when a node is not ready for
-	// executing pods
-	NotReadyTaintTemplate = &v1.Taint{
+	// UnreachableNoScheduleTaintTemplate is the taint with effect NoSchedule
+	// for when a node becomes unreachable.
+	UnreachableNoScheduleTaintTemplate = &v1.Taint{
+		Key:    algorithm.TaintNodeUnreachable,
+		Effect: v1.TaintEffectNoSchedule,
+	}
+
+	// NotReadyNoExecuteTaintTemplate is the taint with effect NoExecute for
+	// when a node is not ready for executing pods
+	NotReadyNoExecuteTaintTemplate = &v1.Taint{
 		Key:    algorithm.TaintNodeNotReady,
 		Effect: v1.TaintEffectNoExecute,
 	}
 
+	// NotReadyNoScheduleTaintTemplate is the taint with effect NoSchedule for
+	// when a node is not ready for executing pods
+	NotReadyNoScheduleTaintTemplate = &v1.Taint{
+		Key:    algorithm.TaintNodeNotReady,
+		Effect: v1.TaintEffectNoSchedule,
+	}
+
 	nodeConditionToTaintKeyMap = map[v1.NodeConditionType]string{
```

Comment:
1. Do not add the `NotReadyNoScheduleTaintTemplate`
1. Change map to `map[v1.NodeCondtionType]struct{TaintKey string, Status v1.ConditionStatus}`
1. Update `doNoScheduleTaintingPass` to check the `Status` instead of `True`

Reply:
Thanks, I simplify the codes according to your comments.